libkbfs/md_journal.go

```diff
@@ -851,7 +851,10 @@ func (j *mdJournal) clear(
 	}
 
 	if head == (ImmutableBareRootMetadata{}) || head.BID() != bid {
-		// Nothing to do.
+		if j.branchID == bid {
```

Comment:
Also, mind adding some unit tests for clear with `bid != j.branchID` and clear with an empty journal but non-zero `j.branchID`?